tickit_devices/synchrotron/synchrotron_topup.py

```diff
@@ -59,12 +75,39 @@ def update(self, time: SimTime, inputs: Inputs) -> DeviceUpdate[Outputs]:
                 The produced update event which contains the value of the machine
                 top up records.
         """
+        new_current = inputs["current"]
+
+        current_difference = new_current - self.last_current
+
+        topup_fill = current_difference > 0
+
+        # calculate the number of updates to go until topup stage begins or
+        # ends: based upon the change of the current.
+        updates_to_go = abs(
+            topup_fill * (300 - new_current) / current_difference
+            + (not topup_fill) * (new_current - 270) / current_difference
+        )
+
+        if self.last_time:
+            self.period = (time - self.last_time) * 1e-9
```

Comment:
Suggestion: I think it makes more sense to have period as just a local variable and set it here in both cases of the `last_time` being None or not

Reply:
Good idea